pkg/apis/core/v1/conversion.go

```diff
@@ -555,3 +563,109 @@ func Convert_core_PersistentVolumeSpec_To_v1_PersistentVolumeSpec(in *core.Persi
 func Convert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in *v1.PersistentVolumeSpec, out *core.PersistentVolumeSpec, s conversion.Scope) error {
 	return autoConvert_v1_PersistentVolumeSpec_To_core_PersistentVolumeSpec(in, out, s)
 }
+
+// handleSeccompVersionSkew implements the version skew behavior described in:
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/20190717-seccomp-ga.md#version-skew-strategy
+func handleSeccompVersionSkew(in *core.Pod, out *v1.Pod) error {
+	klog.V(4).Infof("Handling seccomp version skew fro pod %v", in)
+	annotation, annotationAvailable := in.Annotations[v1.SeccompPodAnnotationKey]
+
+	var field *core.SeccompProfile
+	if in.Spec.SecurityContext != nil {
+		field = in.Spec.SecurityContext.SeccompProfile
+	}
+
+	// If no seccomp annotations or fields are specified, then no further
+	// action is necessary.
+	if field == nil && !annotationAvailable {
+		return nil
+	}
+
+	// TODO: change those to global constants
+	const (
+		profileUnconfined = "unconfined"
+		localhostPrefix   = "localhost/"
+	)
+
+	// If only seccomp fields are specified, add the corresponding annotations.
+	// This ensures that the fields are enforced even if the node version
+	// trails the API version
+	if field != nil && !annotationAvailable {
+		switch field.Type {
+		case core.SeccompProfileUnconfined:
+			out.Annotations[v1.SeccompPodAnnotationKey] = profileUnconfined
+
+		case core.SeccompProfileTypeDefault:
+			out.Annotations[v1.SeccompPodAnnotationKey] = v1.SeccompProfileRuntimeDefault
+
+		case core.SeccompProfileLocalhost:
+			out.Annotations[v1.SeccompPodAnnotationKey] = localhostPrefix + *field.LocalhostProfile
```

Comment:
I am not sure whether it is worth pointing out in comment (or doing the validation) that `field.LocalhostProfile` should not be nil at this point, as there is some validation that blocks that on creation.

Reply:
I'll add a check for that just to be sure. I think we should capture that.